### ISSUE TITLE
ViewportTexture doesn't update (either pink or unresponsive to changes in it's material)

### ISSUE THREAD

**Issue by marcarneg** [godotengine/godot#72616]
### Godot version

4 beta 17

### System information

Windows 10, GTX 970

### Issue description

When rendering a instanced 2D scene as a child of a SubViewport, there are conditions when either the SubViewport-texture stays pink (in my case: on a MeshInstance3D/PlaneMesh/LocalMaterial/Albedo), or won't update changes in the "LocalMaterial"s material (like metallic or roughness). 

Trees:
![grafik](https://user-images.githubusercontent.com/52791267/216401056-0ca999a3-014d-4dbd-b377-3db136e39d20.png)
![grafik](https://user-images.githubusercontent.com/52791267/216401111-a6005ab1-9c45-4672-8323-5450c837b4ff.png)

Some bugs seem to be related to the "Viewport Texture must be set to use it."- error- but not always in consistent ways.
![grafik](https://user-images.githubusercontent.com/52791267/216398403-5cc743ca-a1cf-466c-9580-7f7109e00040.png)

The bugs are occuring on the "Mobile" renderer, independently of the SubViewports "Clear Mode" and "Update Mode".

### Steps to reproduce

1. Create scene 2D
2. add label-control to it
3. (optional: -> add naive @tool-script, which live-updates a rendered string on the label. The one in the example project is naively written, but the editor should handle it nevertheless.)

`@tool
extends Control

@export var thing_to_write : String = "Test" : set = set_thing_to_write
func set_thing_to_write(str : String) -> void:
	thing_to_write = str
	$Panel/Label.text = thing_to_write
`

4. Create scene 3D
5. add SubViewport
6. add 2d scene as instance (child of SubViewport)
7. add MeshInstance3D
8. new PlaneMesh

9. new Material
10. make Material local to scene
11. set Albedo to "NewViewportTexture"
12. set Viewport Path to "SubViewport" 

### Minimal reproduction project

[RenderingportBug.zip](https://github.com/godotengine/godot/files/10571645/RenderingportBug.zip)

**Comment by Zireael07** [comment]
I don't think this method works. However, doing the rough equivalent in script works.

**Comment by marcarneg** [comment]
Not sure what exactly you mean by that- I was able to use this method on another project (where I encountered the bug initially). I successfully projected 2d controls on a 3d plane with exactly this method- it worked, I was able to change the texture via the script until it suddenly broke.

**Comment by vpellen** [comment]
I've encountered various iterations of this issue myself, though in a slightly different form. There's something screwy going on with the way viewport textures are linked in the editor. I think you can do it at runtime through code by calling get_texture on the subviewport, and I think you *might* be able to do that in the editor using a tool script, but this is definitely a recurring issue with viewport textures.

**Comment by Rindbee** [comment]
Similar to #68088, see https://github.com/godotengine/godot/issues/68088#issuecomment-1321067723.
